build/codegen/codegen.sh

```diff
@@ -14,29 +14,58 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-GROUP_VERSIONS="ceph.rook.io:v1"
+##GROUP_VERSIONS="ceph.rook.io:v1"
+
+scriptdir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+basedir="$(cd "${scriptdir}/../.."  && pwd)"
+apisdir="${basedir}"/pkg/apis
+clientdir="${basedir}"/pkg/client
+
+boilerplate="${CODE_GENERATOR}/examples/hack/"boilerplate.go.txt
+kube_codegen="${CODE_GENERATOR}/kube_codegen.sh"
+
+
+echo "DEBUG: scriptdir: ${scriptdir}"
+echo "DEBUG: basedir: ${basedir1}"
+echo "DEBUG: apisdir: ${apisdir}"
+echo "DEBUG: clientdir: ${clientdir}"
+echo "DEBUG: CODE_GENERATOR: ${CODE_GENERATOR}"
+echo "DEBUG: boilerplate ${boilerplate}"
+echo "DEBUG: kube_codegen: ${kube_codegen}"
+ echo "DEBUG: done"
+# exit 0
+
+
+
+set -o pipefail
+
+
+source "${kube_codegen}"
+
+
 
-scriptdir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
 # CODE GENERATION
 # we run deepcopy and client,lister,informer generations separately so we can use the flag "--plural-exceptions"
 # which is only known by client,lister,informer binary and not the deepcopy binary
+#
+
+
 
 # run code deepcopy generation
-bash ${CODE_GENERATOR}/generate-groups.sh \
-    deepcopy \
-    github.com/rook/rook/pkg/client \
-    github.com/rook/rook/pkg/apis \
-    "${GROUP_VERSIONS}" \
-    --output-base "$(dirname "${BASH_SOURCE[0]}")/../../../../.." \
-    --go-header-file "${scriptdir}/boilerplate.go.txt"
+ kube::codegen::gen_helpers \
+    --boilerplate "${boilerplate}" \
+    "${apisdir}"
+
+##    "${GROUP_VERSIONS}" \
 
 # run code client,lister,informer generation
-bash ${CODE_GENERATOR}/generate-groups.sh \
-    client,lister,informer \
-    github.com/rook/rook/pkg/client \
-    github.com/rook/rook/pkg/apis \
-    "${GROUP_VERSIONS}" \
-    --output-base "$(dirname "${BASH_SOURCE[0]}")/../../../../.." \
-    --go-header-file "${scriptdir}/boilerplate.go.txt" \
-    --plural-exceptions "CephNFS:CephNFSes" \
+kube::codegen::gen_client \
+    --output-dir "${basedir}" \
+    --output-pkg "${clientdir}" \
```

Comment:
`--output-pkg` is not a directory per-se. It is a package location with gomod semantics. [Docs](https://github.com/kubernetes/code-generator/blob/b15df6411b47bf6e80bfc63947af6b436b2e05c6/kube_codegen.sh#L400C1-L402C60)

```suggestion
    --output-pkg github.com/rook/rook/pkg/client \
```

Do not use a variable for this. The client dir is (a) not a dir and (b) only used once in the script. Redirection to a var for reading just once only makes the script more complex to read.

Reply:
done